src/v0/destinations/criteo_audience/transform.js

```diff
@@ -0,0 +1,97 @@
+const {
+  BASE_ENDPOINT,
+  operation
+} = require("./config");
+const {
+  defaultRequestConfig,
+  removeUndefinedAndNullValues,
+  simpleProcessRouterDest,
+  defaultPatchRequestConfig
+} = require("../../util");
+const { InstrumentationError } = require("../../util/errorTypes");
+
+const { getAccessToken, preparePayload } = require("./util");
+
+const prepareResponse = (payload, audienceId, accessToken) => {
+  const response = defaultRequestConfig();
+  response.endpoint = `${BASE_ENDPOINT}audiences/${audienceId}/contactlist`;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  response.method = defaultPatchRequestConfig.requestMethod;
+  response.headers = {
+    Authorization: `Bearer ${accessToken}`,
+    "Content-Type": "application/json",
+    Accept: "application/json"
+  };
+  return response;
+};
+/**
+ * This function is used for building the final response to be returned.
+ * @param {*} message
+ * @param {*} destination
+ * @returns
+ */
+const responseBuilder = async (message, destination, accessToken) => {
+  const responseArray = [];
+  const { Config } = destination;
+  const { audienceId } = Config;
+  const { listData } = message.properties;
+  operation.forEach(op => {
+    if (listData[op]) {
+      const criteoPayload = preparePayload(
+        listData[op],
+        op,
+        Config
+      );
+      responseArray.push(
+        prepareResponse(criteoPayload, audienceId, accessToken)
+      );
+    }
+  });
+
+  if (responseArray.length === 0) {
+    throw new InstrumentationError(`Payload could not be populated`);
+  }
+  if (responseArray.length === 1) {
```

Comment:
Here I'm seeing when responseArray contains one element in the array, we're returning the element but if it has multiple elements we're returning `array<element>`.
We can directly send `array<element>` regardless of the number of elements in the array. Any reason for this approach ?

Reply:
@ujjwal-ab would be the right person for this.

Reply:
There is no specific reason. We can directly return the array of responses in both cases that shouldn't cause any issues.

Reply:
Did we verify if we're not having a problem ?

Reply:
Yes, We have verified.